src/cmdstan/io/json/json_data.hpp

```diff
@@ -119,6 +120,43 @@ class json_data : public stan::io::var_context {
     return empty_vec_r_;
   }
 
+  /**
+   * Read out a vector of complex numbers from the specifed name.
+   * @param name Name of Variable.
+   */
+  std::vector<std::complex<double>> vals_c(const std::string &name) const {
+    if (contains_r_only(name)) {
+      auto &&vec_r = (vars_r_.find(name)->second);
+      auto &&val_r = vec_r.first;
+      auto &&dim_r = vec_r.second;
+      std::vector<std::complex<double>> vec_c(val_r.size() / 2);
+      int offset = 1;
+      for (int i = 0; i < dim_r.size() - 1; ++i) {
+        offset *= dim_r[i];
+      }
+      for (int i = 0; i < vec_c.size(); ++i) {
+        vec_c[i] = std::complex<double>{static_cast<double>(val_r[i]),
```

Comment:
Why does `val_r[i]` need to be cast here?   Isn't it already a `double` value?  And even if it's not a double value, calling the `std::complex<double>` constructor should automatically cast it to `double` because of the template specification.